scripts/fill_dcp1_file_metadata/fill_dcp1_metadata.py

```diff
@@ -0,0 +1,188 @@
+
+    return uuid_metadata_map_filtered
+
+
+def set_up_ingestapi() -> IngestApi:
+    """
+    Set up ingestApi object, with credentials to access and modify information in the database. URL is not asked because
+    this script can only be run in production (Azul service not available in dev/staging)
+    :return ingest_api: IngestApi object set up with token
+    """
+    ingest_api = IngestApi("https://api.ingest.archive.data.humancellatlas.org")
+    ingest_api.set_token(f"Bearer {INGEST_TOKEN}")
+    return ingest_api
+
+
+def patch_file_metadata(uuid_metadata_map: dict, ingest_api: IngestApi, dry_run: bool):
+    """
+    Send a PUT request to each of the files in the uuid_size_map with the updated size and fileContentType extracted
+    from Azul
+    :param uuid_metadata_map:   {file_uuid: metadata} map of all the files that need to be updated
+    :param ingest_api:          IngestApi object with headers set up
+    :param dry_run:             bool value, if True no PUT operations will be performed (But information will be logged)
+    :return:
+    """
+    for uuid, metadata in uuid_metadata_map.items():
+        # Check file exists in Ingest (Non-organically described matrices are virtually indistinguishable via metadata)
+        try:
+            logging.info(f"Searching for file metadata for file {metadata['name']}")
+            entity = ingest_api.get_entity_by_uuid('files', uuid)
+        except HTTPError:
+            logging.warning(f"Could not find file {metadata['name']} in ingest, please make sure it's not a "
+                            f"non-organically described matrix.More details:")
+            logging.warning(json.dumps(metadata, indent=4, separators=(", ", ": ")))
+            continue
+
+        # Check entity is not already filled in - Script will only fill size and fileContentType when both are missing
+        if entity.get('size') and entity.get('fileContentType'):
+            logging.warning(f"File {entity.get('fileName')}, with UUID {uuid} already has size and fileContentType set")
+            continue
+
+        # Add size and fileContentType to metadata
+        entity['size'] = metadata['size']
+        data_type = f"{'application/gzip' if metadata['format'] == 'fastq.gz' else 'application/octet-stream'}; dcp-type=data"
+        entity['fileContentType'] = data_type
+
+        logging.info(f"PATCH file {entity.get('fileName')}, uuid {uuid}, with size {entity['size']} and dcp-type {entity['fileContentType']}")
+
+        if dry_run:
+            continue
+
+        # PUT the metadata back with the newly introduced values
+        ingest_api.put(entity['_links']['self']['href'], json=entity)
+        logging.info(f"PATCHED file {entity.get('fileName')}, url: {entity['_links']['self']['href']}")
+
+
+def main(project_uuid, dry_run):
+    # Set-up logging + early messages/warnings
+    logging.basicConfig(filename=f'{project_uuid}.log', level=logging.INFO, filemode='w',
```

Comment:
logging level should also be configurable. possibly as an environment variable.

Reply:
Can I get some info on how it should be configurable? what are the parameters that should the user decide? I have little to 0 experience with logging

Reply:
simplest way: `os.environ['LOG_LEVEL']`